--- testdisk-5.4-WIP/src/types.h.orig	2004-08-22 11:10:40.000000000 +0200
+++ testdisk-5.4-WIP/src/types.h	2004-08-27 22:51:55.555552936 +0200
@@ -25,7 +25,9 @@
 #include <stdint.h>
 #endif
 #include <sys/types.h>
-#ifdef HAVE_NCURSES_H
+#ifdef HAVE_NCURSES_NCURSES_H
+#include <ncurses/ncurses.h>
+#elif defined(HAVE_NCURSES_H)
 #include <ncurses.h>
 #else
 #ifdef HAVE_CURSES_H
--- testdisk-5.4-WIP/configure.ac.orig	2004-08-25 19:12:59.000000000 +0200
+++ testdisk-5.4-WIP/configure.ac	2004-08-27 22:52:53.475747728 +0200
@@ -143,7 +143,7 @@
 
 # Checks for header files.
 AC_HEADER_STDC
-AC_CHECK_HEADERS([ byteswap.h curses.h dal/file_dal.h dal/file.h endian.h fcntl.h features.h linux/fs.h linux/hdreg.h locale.h machine/endian.h ncurses.h signal.h stdint.h stdlib.h string.h sys/disk.h sys/disklabel.h sys/dkio.h sys/endian.h sys/ioctl.h sys/mount.h sys/param.h sys/stat.h sys/time.h sys/vtoc.h time.h unistd.h])
+AC_CHECK_HEADERS([ byteswap.h curses.h dal/file_dal.h dal/file.h endian.h fcntl.h features.h linux/fs.h linux/hdreg.h locale.h machine/endian.h ncurses.h ncurses/ncurses.h signal.h stdint.h stdlib.h string.h sys/disk.h sys/disklabel.h sys/dkio.h sys/endian.h sys/ioctl.h sys/mount.h sys/param.h sys/stat.h sys/time.h sys/vtoc.h time.h unistd.h])
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
