--- testdisk/src/Makefile.orig	2004-01-10 13:06:13.000000000 +0000
+++ testdisk/src/Makefile	2004-03-06 21:19:37.000000000 +0000
@@ -9,11 +9,11 @@
 VER_E2FSPROGS=1.34
 VER_PROGSREISERFS=0.3.1-rc8
 VER_NTFSPROGS=1.8.2
-LIBREISER=../progsreiserfs-$(VER_PROGSREISERFS)/libreiserfs/.libs/libreiserfs.a
-INCREISER=../progsreiserfs-$(VER_PROGSREISERFS)/include
-LIBEXT=../e2fsprogs-$(VER_E2FSPROGS)/lib/ext2fs/libext2fs.a
-INCNTFS=../ntfsprogs-$(VER_NTFSPROGS)
-LIBNTFS=../ntfsprogs-$(VER_NTFSPROGS)/libntfs/.libs/libntfs.a
+LIBREISER=-lreiserfs
+INCREISER=/usr/include
+LIBEXT=-lext2fs
+INCNTFS=/usr/include/ntfs
+LIBNTFS=-lntfs
 CFLAGS=-Wall -MD -Wpointer-arith -Wmissing-declarations -Wmissing-prototypes -Wstrict-prototypes -Wshadow -Wwrite-strings -W -Wcast-align -Waggregate-return -Wbad-function-cast -Wcast-qual -Wundef -Wredundant-decls -Wsign-compare -O2 -g -I. -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -Wnested-externs -Winline -DHAVE_E2FSPROGS -DHAVE_PROGSREISERFS31 -DHAVE_NTFSPROGS
 #-DHAVE_PROGSREISERFS30
 # Debian uses progsreiserfs 3.0
--- testdisk/src/Makefile.orig	2004-03-06 21:23:54.000000000 +0000
+++ testdisk/src/Makefile	2004-03-06 21:24:50.000000000 +0000
@@ -72,7 +72,7 @@
 	echo \#define COMPILE_HOST \"`hostname`\" >> version.h
 	$(MAKE) testdisk CFLAGS="$(CFLAGS) -fms-extensions -DBSD -DHAVE_SNPRINTF -DHAVE_VSNPRINTF -pipe"  LIBS="$(LIBS) -lncurses"
 
-testdisk: $(LIBSA) $(OBJS) 
+testdisk: $(OBJS) 
 	$(CC) $(CFLAGS) -o testdisk $(OBJS) $(LIBS) $(LIBSA) $(LDFLAGS)
 
 testdisk_static: $(LIBSA) $(OBJS) 
--- testdisk/src/types.h.org	2004-01-09 16:17:56.000000000 +0000
+++ testdisk/src/types.h	2004-03-06 21:27:49.000000000 +0000
@@ -19,7 +19,7 @@
     foundation, inc., 675 mass ave, cambridge, ma 02139, usa.
 
  */
-#include <sys/types.h>
+#include <linux/types.h>
 #ifdef DJGPP
 #include <bios.h>
 #include <curses.h>
